Sources/SwiftFusion/Inference/ChordalInitialization.swift

```diff
@@ -0,0 +1,213 @@
+// Copyright 2020 The SwiftFusion Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import PenguinStructures
+import TensorFlow
+
+/// A BetweenFactor alternative that uses the Chordal (Frobenious) norm on rotation for Rot3
+/// Please refer to Carlone15icra (Initialization Techniques for 3D SLAM: a Survey on Rotation Estimation and its Use in Pose Graph Optimization)
+/// for explanation.
+public struct FrobeniusFactorRot3: LinearizableFactor
+{
+  public typealias Variables = Tuple2<Vector9, Vector9>
+  public typealias JacobianRows = Array9<Tuple2<Vector9.TangentVector, Vector9.TangentVector>>
+  
+  public let edges: Variables.Indices
+  public let difference: Vector9
+
+  public init(_ startId: TypedID<Vector9, Int>, _ endId: TypedID<Vector9, Int>, _ difference: Vector9) {
```

Comment:
I don't love "start" and "end" here, just say id1, id2?

Reply:
done

Reply:
done